server/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -353,4 +354,18 @@ default Set<String> getUsageFeatures() {
     static boolean assertSnapshotMetaThread() {
         return ThreadPool.assertCurrentThreadPool(ThreadPool.Names.SNAPSHOT_META);
     }
+
+    /**
+     * Get the current count of snapshots in progress
+     *
+     * @return The current number of shard snapshots in progress metric value, or null if this repository doesn't track that
+     */
+    @Nullable
+    default LongWithAttributes getShardSnapshotsInProgress() {
+        return null;
+    }
+
+    default RepositoriesStats.SnapshotStats getSnapshotStats() {
+        return new RepositoriesStats.SnapshotStats(getRestoreThrottleTimeInNanos(), getSnapshotThrottleTimeInNanos());
```

Comment:
Likewise here, can we not have a `default` implementation and instead push this down to the subclasses? We should be able to drop `getRestoreThrottleTimeInNanos()` and `getSnapshotThrottleTimeInNanos()` from the `Repository` interface with this change.